deflate.c

```diff
@@ -620,13 +621,13 @@ int32_t ZEXPORT PREFIX(deflateParams)(PREFIX3(stream) *strm, int32_t level, int3
     if (level < 0 || level > 9 || strategy < 0 || strategy > Z_FIXED) {
         return Z_STREAM_ERROR;
     }
-    DEFLATE_PARAMS_HOOK(strm, level, strategy);  /* hook for IBM Z DFLTCC */
+    DEFLATE_PARAMS_HOOK(strm, level, strategy, &hook_flush);  /* hook for IBM Z DFLTCC */
     func = configuration_table[s->level].func;
 
-    if ((strategy != s->strategy || func != configuration_table[level].func) &&
+    if ((strategy != s->strategy || func != configuration_table[level].func || hook_flush != Z_NO_FLUSH) &&
         s->last_flush != -2) {
         /* Flush the last buffer: */
-        int err = PREFIX(deflate)(strm, Z_BLOCK);
+        int err = PREFIX(deflate)(strm, RANK(hook_flush) > RANK(Z_BLOCK) ? hook_flush : Z_BLOCK);
```

Comment:
I think this deserves a comment to avoid future confusion, since it is not obvious.
Alternately an if-else block with comment.

Reply:
ok